extensions/wikia/UserLogin/tests/UserLoginHelperTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+class UserLoginHelperTest extends WikiaBaseTest {
+
+	/**
+	 * @dataProvider getNewAuthUrlDataProvider
+	 */
+	public function testGetNewAuthUrl( $page, $requestUrl, $contLangCode, $expected ) {
+		$titleMock = $this->getMock( Title::class, [ 'isSpecial' ] );
+		$titleMock->expects( $this->once() )->method( 'isSpecial' )->willReturn( false );
+
+		$webRequestMock = $this->getMock( WebRequest::class, [ 'getRequestURL' ] );
+		$webRequestMock->expects( $this->once() )->method( 'getRequestURL' )->willReturn( $requestUrl );
+
+		$languageMock = $this->getMock( Language::class );
+		$languageMock->mCode = $contLangCode;
+
+		$this->mockGlobalVariable( 'wgTitle', $titleMock );
+		$this->mockGlobalVariable( 'wgRequest', $webRequestMock );
+		$this->mockGlobalVariable( 'wgContLang', $languageMock );
+
+		$userLoginHelper = new UserLoginHelper();
+
+		$authUrl = $userLoginHelper->getNewAuthUrl( $page );
+
+		$this->assertEquals( $expected, $authUrl );
+	}
+
+	public function getNewAuthUrlDataProvider() {
+		return [
+			[ '/join', 'http://www.wikia.com', 'en', '/join?redirect=http%3A%2F%2Fwww.wikia.com' ],
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "http://www.wikia.com" 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)